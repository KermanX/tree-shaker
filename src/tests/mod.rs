use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
