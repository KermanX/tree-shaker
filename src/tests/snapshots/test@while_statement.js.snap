---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/while_statement.js
---
export function main() {
	while (effect);
	while (sth) {
		effect1;
		break;
	}
	while (sth) {
		effect1;
		continue;
	}
	while (sth) {
		effect1;
		return;
	}
	while (sth1) {
		effect1;
		while (sth2) {
			effect2;
			break;
		}
		effect2;
	}
	while (1) {
		effect1;
		return;
	}
	outer: while (a) {
		effect1;
		while (b) {
			effect2;
			break outer;
		}
		effect3;
	}
	while (sth()) {
		if (a) {
			break;
		} else {
			if (b) {
				break;
			} else {
				continue;
			}
		}
	}
	while (sth()) {
		i: if (a) {
			break i;
		} else {
			if (b) {
				break;
			} else {
				continue;
			}
		}
		effect2;
	}
}
